### ISSUE TITLE
Support multiline format

### ISSUE THREAD

**Issue by moritzheiber** [sonots/fluent-plugin-tail_path#3]
I'd like to have `multiline` format support within this plugin. Right now, when I try to run this plugin using the `multiline` format type I get this:

```
2015-06-09 12:45:16 +0000 [error]: undefined method `parser' for #<Fluent::TextParser::MultilineParser:0x00000002271af8>
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/bundler/gems/fluent-plugin-tail_path-88a30118e2ba/lib/fluent/plugin/in_tail_path.rb:104:in `block in parse_multilines'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/bundler/gems/fluent-plugin-tail_path-88a30118e2ba/lib/fluent/plugin/in_tail_path.rb:103:in `each'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/bundler/gems/fluent-plugin-tail_path-88a30118e2ba/lib/fluent/plugin/in_tail_path.rb:103:in `parse_multilines'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:222:in `call'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:222:in `receive_lines'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:322:in `call'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:322:in `wrap_receive_lines'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:514:in `call'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:514:in `on_notify'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:347:in `on_notify'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:448:in `call'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:448:in `on_change'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/cool.io-1.3.1/lib/cool.io/loop.rb:88:in `run_once'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/cool.io-1.3.1/lib/cool.io/loop.rb:88:in `run'
  2015-06-09 12:45:16 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:215:in `run'
2015-06-09 12:45:17 +0000 [error]: closed stream
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:497:in `read_nonblock'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:497:in `on_notify'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:347:in `on_notify'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:432:in `call'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:432:in `on_timer'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/cool.io-1.3.1/lib/cool.io/loop.rb:88:in `run_once'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/cool.io-1.3.1/lib/cool.io/loop.rb:88:in `run'
  2015-06-09 12:45:17 +0000 [error]: bundle/ruby/2.0/gems/fluentd-0.12.11/lib/fluent/plugin/in_tail.rb:215:in `run'
2015-06-09 12:45:18 +0000 [error]: closed stream
```

Configuration file would be:

```
<source>
  type tail_path
  format multiline
  format_firstline /^\d{4}-\d{1,2}-\d{1,2} \d{1,2}:\d{1,2}:\d{1,2},\d{1,3}.*/
  format1 /^(?<meta.timestamp>\d{4}-\d{1,2}-\d{1,2} \d{1,2}:\d{1,2}:\d{1,2},\d{1,3})(?<message>.*)/
  path /tmp/application.log
  refresh_interval 5
  pos_file /tmp/file.pos
  path_key /tmp/application.log
  tag logs
</source>    

<match logs>
  type stdout  
  log_level trace
</match>
```

The expected outcome would obviously be that the `path` key gets added to the output. PR #2 tries to implements this, but the tests fail and my initial tests didn't find any improvements.

**Comment by llchen223** [comment]
I need this too!

**Comment by llavaud** [comment]
Yes, me too ! please ! ;)

**Comment by sonots** [comment]
Should be fixed by v0.1.0. Please try.
